src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -134,7 +175,34 @@ public void Dispose()
             }
         }
 
-        private async Task<string> GetLastPollStatusAsync()
+        private async Task MoveToPoisonQueueAsync(ApiHubFile apiHubFile)
+        {
+            CloudQueueClient queueClient = CloudStorageAccount.Parse(this._config.StorageConnectionString).CreateCloudQueueClient();
```

Comment:
You should do this once on construction, or otherwise find a way to cache the queue client